README.md

```diff
@@ -6,116 +6,200 @@ A hook to abstract the more complicated wiring of web-components inside of react
 [![Package Downloads](https://img.shields.io/npm/dm/use-web-component.svg)](https://npmjs.org/package/use-web-component)
 ![CI Status](https://github.com/kaiyote/use-web-component/workflows/Pull%20Request%20Test/badge.svg)
 
-## Installation
+- [Installation](#Installation)
+- [Purpose](#Purpose)
+- [Usage](#Usage)
+  - [Higher Order Component](#higher-order-component)
+  - [Hook](#hook)
+- [Examples](#Examples)
+
+# Installation
 `npm install use-web-component`
 
-## Type Signature
+# Purpose
+When it comes to custom elements / web components, `React` doesn't support anything other than standard HTML event binding (`onClick`, etc) and simple (string, number, bool) attributes. If you need to attach a listener for a custom event, or a callback function as a property, you're forced to attach a ref, and touch it in a `useEffect` or `useLayoutEffect` to setup everything.
+
+`useWebComponent` was designed to hide all of that from you, so all you have do is give it your props (not necessarily all of them) and it handles wiring custom events, setting properties, and then returns any "simple" attributes you may have passed it as the first element of its return tuple so you can spread those onto the element yourself, as well as a ref that you attach to the jsx tag so it all gets wired correctly.
+
+Early user testing suggested that having to deal with the ref yourself was less than ideal, so the `withWebComponent` HoC was created. With it all you do is tell it the tag name of the web component, and it generates a React Component for you that already does the ref wiring inside.
+
+# Usage
+## Higher Order Component
+### Type Signature
+```ts
+function withWebComponent (tagName: string): ForwardRefExoticComponent
+```
+- `tagName` is literally the string name of the web-component tag.
+- the `HoC` returns a ReactComponent that will correctly forward refs so you can just use it as you would a regular react component
+  - the `HoC` component also has two extra props that are the second and third arguments of the hook, so you can configure prop name mapping and whether or not the events are published as camel case
+
+### Basic Usage
+```jsx
+import { withWebComponent } from 'use-web-component'
+
+function App () {
+  const MyComponent = withWebComponent('my-component')
+  return (
+    <MyComponent
+      simple-attr='this passed through as an attribute'
+      complexProp={{ value: 'this will be set as a property on the tag' }}
+      onCustomEvent={e => { console.log('customEvent happened') }}
+      eventsAreCamelCase
+    />
+  )
+}
+
+/* renders <my-component simple-attr='this passed through as an attribute'></my-component> with a listener bound to 'customEvent' and a property of 'complexProp' set to { value: 'this will be set as a property on the tag' } */
```

Comment:
Turns out github doesn't linewrap in code blocks (which makes sense) so we do actually need a hard newline here.
```suggestion
/* renders <my-component simple-attr='this passed through as an attribute'></my-component> with a listener bound to 
'customEvent' and a property of 'complexProp' set to { value: 'this will be set as a property on the tag' } */
```